src/daft-image/src/functions/hash.rs

```diff
@@ -0,0 +1,102 @@
+    prelude::*,
+};
+use daft_dsl::{
+    ExprRef,
+    functions::{FunctionArgs, ScalarUDF},
+};
+use serde::{Deserialize, Serialize};
+
+#[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Eq, Hash)]
+pub struct ImageHash;
+
+#[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Eq, Hash)]
+pub enum ImageHashAlgorithm {
+    Average,
+    Perceptual,
+    Difference,
+    Wavelet,
+    CropResistant,
+}
+
+impl std::str::FromStr for ImageHashAlgorithm {
+    type Err = DaftError;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        match s.to_lowercase().as_str() {
+            "average" => Ok(Self::Average),
+            "perceptual" => Ok(Self::Perceptual),
+            "difference" => Ok(Self::Difference),
+            "wavelet" => Ok(Self::Wavelet),
+            "crop_resistant" => Ok(Self::CropResistant),
+            _ => Err(DaftError::ValueError(format!(
+                "Invalid image hash algorithm: {}. Must be one of: average, perceptual, difference, wavelet, crop_resistant",
+                s
+            ))),
+        }
+    }
+}
+
+impl FromLiteral for ImageHashAlgorithm {
+    fn try_from_literal(lit: &Literal) -> DaftResult<Self> {
+        match lit {
+            Literal::Utf8(s) => s.parse(),
+            _ => Err(DaftError::TypeError(format!(
+                "Expected string literal for image hash algorithm, got: {:?}",
+                lit
+            ))),
+        }
+    }
+}
+
+#[typetag::serde]
+impl ScalarUDF for ImageHash {
+    fn call(&self, inputs: FunctionArgs<Series>) -> DaftResult<Series> {
+        let input = inputs.required((0, "input"))?;
+        let algorithm_series = inputs.required((1, "algorithm"))?;
+        
+        // Extract the algorithm from the series (should be a scalar)
+        let algorithm_str = algorithm_series.utf8()?.get(0).ok_or_else(|| {
+            DaftError::ValueError("algorithm must be a scalar string".to_string())
+        })?;
+        
+        let algorithm: ImageHashAlgorithm = algorithm_str.parse()?;
+        
+        match algorithm {
+            ImageHashAlgorithm::Average => crate::series::average_hash(input),
+            ImageHashAlgorithm::Perceptual => crate::series::perceptual_hash(input),
+            ImageHashAlgorithm::Difference => crate::series::difference_hash(input),
+            ImageHashAlgorithm::Wavelet => crate::series::wavelet_hash(input),
+            ImageHashAlgorithm::CropResistant => crate::series::crop_resistant_hash(input),
+        }
+    }
+
+    fn name(&self) -> &'static str {
+        "image_hash"
+    }
+
+    fn get_return_field(
+        &self,
+        inputs: FunctionArgs<ExprRef>,
+        schema: &Schema,
+    ) -> DaftResult<Field> {
+        let input = inputs.required((0, "input"))?;
+        let field = input.to_field(schema)?;
+
+        match field.dtype {
+            DataType::Image(_) | DataType::FixedShapeImage(..) => {
+                Ok(Field::new(field.name, DataType::Utf8))
```

Comment:
as suggested elsewhere, I believe we can just return `DataType::Uint64` instead of a Utf8